dali/c_api/c_api_test.c

```diff
@@ -0,0 +1,16 @@
+// Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "dali/c_api.h" 
```

Comment:
Would be nice to actually build a pipeline (one valid, and one that reports an error).

Reply:
Not so easy unfortunately:
1. Can't generate any Pipeline in C code
2. Can't use GTest in C code

Reply:
> Not so easy unfortunately:
> 
>     1. Can't generate any Pipeline in C code

You don't have to, you can take a simple Pipeline, serialize it, and use that string for test. For ease of recreating the Pipeline if anything changes, I would add this as a simple python script in some utils directory and point to it with a comment.

>     2. Can't use GTest in C code

Yes, that's true, you would need to check it by hand. As the C API is rather small, I would stick with keeping the test quite simple but you can take a look at https://en.wikipedia.org/wiki/List_of_unit_testing_frameworks#C